### ISSUE TITLE
default parameters for errata_tool_user are not idempotent

### ISSUE THREAD

**Issue by ktdreyer** [ktdreyer/errata-tool-ansible#80]
The following is not idempotent:

```yaml
- name: Add developer Errata Tool account
  errata_tool_user:
    login_name: cooldeveloper@redhat.com
    realname: Cool Developer
```

On the second run, Ansible says we're changing some things:
```
  - changing receives_mail from False to True
  - changing roles from [] to None
```

**Comment by ktdreyer** [comment]
The `roles` problem is simple: we need to default to an empty list.
```diff
diff --git a/library/errata_tool_user.py b/library/errata_tool_user.py
index 8fdaf1e..0d9a401 100644
--- a/library/errata_tool_user.py
+++ b/library/errata_tool_user.py
@@ -168,7 +168,7 @@ def run_module():
         realname=dict(required=True),
         organization=dict(),
         receives_mail=dict(type='bool', default=True),
-        roles=dict(type='list'),
+        roles=dict(type='list', default=[]),
         enabled=dict(type='bool', default=True),
         email_address=dict(),
     )
```

For `receives_mail`, I can't see how we're ending up with "receives_mail" as `False` here. Might need to narrow this down through a `curl` reproducer script.

**Comment by ktdreyer** [comment]
I found the problem with "receives_mail" and filed CLOUDWF-2817. To summarize that ticket: When `finger` fails to find a user account and I POST or PUT to `/api/v1/user`, the ET almost always resets the user's "`receives_mail`" attribute back to "`false`" (even when I set it to "`true`"). The only way to override this behavior is to send along the "`email_address`" parameter in every POST or PUT call.
